src/netex_idf/stops.rs

```diff
@@ -14,65 +14,89 @@
 // along with this program.  If not, see
 // <http://www.gnu.org/licenses/>.
 
+use super::EUROPE_PARIS_TIMEZONE;
 use crate::{
     minidom_utils::{TryAttribute, TryOnlyChild},
     model::Collections,
-    objects::{Codes, Coord, Properties, StopArea},
+    objects::{Codes, Coord, Properties, StopArea, StopPoint, StopType},
     Result,
 };
-use failure::{format_err, ResultExt};
+use failure::{bail, format_err, ResultExt};
 use log::{info, warn};
 use minidom::Element;
+use proj::Proj;
 use std::{collections::HashMap, fs::File, io::Read};
 use transit_model_collection::CollectionWithId;
 
-fn load_stop_areas(elem: &Element) -> Result<CollectionWithId<StopArea>> {
-    let mut stop_areas = CollectionWithId::default();
-    let members = elem
-        .try_only_child("dataObjects")?
-        .try_only_child("CompositeFrame")?
-        .try_only_child("frames")?
-        .children()
-        .flat_map(|e| e.children())
-        .filter(|e| e.name() == "members");
+fn load_stop_area(stop_place_elem: &Element, id: String) -> Result<StopArea> {
+    // LDA or ZDL without ParentSiteRef
+    let mut stop_area = StopArea {
+        id,
+        name: stop_place_elem
+            .try_only_child("Name")?
+            .text()
+            .trim()
+            .to_string(),
+        visible: true,
+        coord: Coord { lon: 0., lat: 0. },
+        ..Default::default()
+    };
 
-    let mut map_lda_zde: HashMap<String, String> = HashMap::default();
-    for member in members {
-        let stop_places = member.children().filter(|e| e.name() == "StopPlace");
+    // add object properties
+    let type_of_place_ref: String = stop_place_elem
+        .try_only_child("placeTypes")?
+        .try_only_child("TypeOfPlaceRef")?
+        .try_attribute("ref")?;
+    stop_area
```

Comment:
Maybe you can create the `KeyValues` instance first, then create the `StopArea`, the `stop_area` would not need to be `mut`.

Reply:
done